packages/yew-macro/src/props/component.rs

```diff
@@ -156,17 +86,30 @@ impl ComponentProps {
                         .build()
                 }
             }
-            Self::With(with_props) => {
+            // Builder pattern is unnecessary in this case, since the base expression guarantees
+            // all values are initialized
+            Some(expr) => {
                 let ident = Ident::new("__yew_props", props_ty.span());
+                let set_props = self.props.iter().map(|Prop { label, value, .. }| {
+                    if is_implicitly_converted(value) {
+                        quote_spanned! {value.span()=>
+                            #ident.#label = #value.into();
+                        }
+                    } else {
```

Comment:
Nit: 
I think `is_implicitly_converted` doesn't describe the fn but the action taken if the fn returns `true`. I think something like `is_lit_str` would be more clear for the fn then a comment explaining the conversion inside the `if` block.

Reply:
Makes sense. The reason I named it as such was in case there were other literals that we would want to implicitly convert, but I expect that string literals are the only kind that will often require a conversion

Reply:
A fn that does the conversion if required returning the value tokens could be an alternative approach that would allow more flexibility and would remove the need for the if/else block?

Reply:
That was my initial approach, but in this case, "performing the conversion" just means appending `.into()` onto the assignment, and the code ended up less readable than doing it this way. I think I'll just change the name as you initially suggested